core/src/mast/mod.rs

```diff
@@ -528,6 +526,18 @@ impl MastNodeId {
         Self::from_u32_with_node_count(value, mast_forest.nodes.len())
     }
 
+    pub fn from_usize_safe(
+        node_id: usize,
+        mast_forest: &MastForest,
+    ) -> Result<Self, DeserializationError> {
+        let node_id: u32 = node_id.try_into().map_err(|_| {
+            DeserializationError::InvalidValue(format!(
+                "Invalid node id '{node_id}' while deserializing"
```

Comment:
```suggestion
                "node id '{node_id}' does not fit into a u32"
```
Nit: Improve error so it's more helpful for debugging.